lib/src/chart_painter.dart

```diff
@@ -51,58 +54,59 @@ class PieChartPainter extends CustomPainter {
       }
       _paintList.add(paint);
     }
-    _totalAngle = angleFactor * math.pi * 2;
     _subParts = values;
     _subTitles = titles;
-    _total = values.fold(0, (v1, v2) => v1 + v2);
   }
 
   @override
   void paint(Canvas canvas, Size size) {
     final side = size.width < size.height ? size.width : size.height;
-    _prevAngle = this.initialAngle*math.pi/180;
-    for (int i = 0; i < _subParts.length; i++) {
+    if (emptyColor != null && _total == 0) {
```

Comment:
Do not explicitly check emptyColor value. Just make emptyColor having a default Color in constructor

Reply:
Fixed.